pkg/api/client_test.go

```diff
@@ -71,3 +71,153 @@ func TestState(t *testing.T) {
 		})
 	}
 }
+
+func TestSetState(t *testing.T) {
+	for i, tc := range []struct {
+		Input  map[string]*State
+		Output map[string]*State
+	}{
+		{
+			Input: map[string]*State{
+				"foo": {
+					ID:  "1",
+					CPB: CPBNo,
+				},
+				"bar": {
+					ID:           "2",
+					RefBoxRole:   RoleDefenderAssist,
+					Kinect2State: KinectStateBall,
+				},
+				"baz": {
+					ID:           "3",
+					ActiveDevPC:  uint8(42),
+					Temperature1: uint8(69),
+				},
+			},
+			Output: map[string]*State{
+				"foo": {
+					ID:  "1",
+					CPB: CPBYes,
+				},
+				"bar": {
+					ID:           "2",
+					RefBoxRole:   RoleDefenderAssist2,
+					Kinect2State: KinectStateNoBall,
+				},
+				"baz": {
+					ID:           "3",
+					ActiveDevPC:  uint8(89),
+					Temperature1: uint8(69),
+				},
+			},
+		},
+	} {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			a := assert.New(t)
+
+			var writerExec bool //used for checking whether the writer is actually executed
+
+			out := &bytes.Buffer{}
+			in := &mockWriter{
+				WriteFunc: func(b []byte) (int, error) {
+					writerExec = true
```

Comment:
Shouldn't this be placed at the end of the function?

Reply:
No